core/src/main/scala/kafka/log/Log.scala

```diff
@@ -478,7 +478,7 @@ class Log(@volatile var dir: File,
     completedTxns.foreach(producerStateManager.completeTxn)
   }
 
-  private[log] def activePids: Map[Long, ProducerIdEntry] = lock synchronized {
+  private[log] def activeProducers: Map[Long, ProducerIdEntry] = lock synchronized {
```

Comment:
Not from this PR, but it seems like `updateProducers` should just return the result from `appendInfo.append(batch)` instead of taking a list to append to (from a simplicity perspective).